test/controllers/ChoiceControllerSpec.scala

```diff
@@ -0,0 +1,134 @@
+        givenTheCacheIsEmpty()
+
+        val result = controller(SuccessfulAuth()).displayChoiceForm(get)
+
+        status(result) mustBe Status.OK
+        contentAsHtml(result) mustBe page(Choice.form())
+      }
+
+      "existing answers" in {
+        givenTheCacheContains(Cache("pid", ArrivalAnswers()))
+
+        val result = controller(SuccessfulAuth()).displayChoiceForm(get)
+
+        status(result) mustBe Status.OK
+        contentAsHtml(result) mustBe page(Choice.form().fill(Choice.Arrival))
+      }
+    }
+
+    "return 403 when unauthenticated" in {
+      val result = controller(UnsuccessfulAuth).displayChoiceForm(get)
+
+      status(result) mustBe Status.FORBIDDEN
+    }
+  }
+
+  "POST" should {
+
+    "return 200 when authenticated" when {
+      "arrival" in {
+        val post = FakeRequest("POST", "/").withFormUrlEncodedBody("choice" -> Choice.Arrival.value).withCSRFToken
+        val result = controller(SuccessfulAuth()).submitChoice(post)
+
+        status(result) mustBe Status.SEE_OTHER
+        redirectLocation(result) mustBe Some(routes.ChoiceController.displayChoiceForm().url)
+        theCacheUpserted mustBe Cache(pid, ArrivalAnswers())
+      }
+
+      "departure" in {
+        val post = FakeRequest("POST", "/").withFormUrlEncodedBody("choice" -> Choice.Departure.value).withCSRFToken
+        val result = controller(SuccessfulAuth()).submitChoice(post)
+
+        status(result) mustBe Status.SEE_OTHER
+        redirectLocation(result) mustBe Some(routes.ChoiceController.displayChoiceForm().url)
+        theCacheUpserted mustBe Cache(pid, DepartureAnswers())
+      }
+
+      "associate UCR" in {
+        val post = FakeRequest("POST", "/").withFormUrlEncodedBody("choice" -> Choice.AssociateUCR.value).withCSRFToken
+        val result = controller(SuccessfulAuth()).submitChoice(post)
+
+        status(result) mustBe Status.SEE_OTHER
+        redirectLocation(result) mustBe Some(routes.ChoiceController.displayChoiceForm().url)
+        theCacheUpserted mustBe Cache(pid, AssociateUcrAnswers())
+      }
+
+      "disassociate UCR" in {
+        val post = FakeRequest("POST", "/").withFormUrlEncodedBody("choice" -> Choice.DisassociateUCR.value).withCSRFToken
+        val result = controller(SuccessfulAuth()).submitChoice(post)
+
+        status(result) mustBe Status.SEE_OTHER
+        redirectLocation(result) mustBe Some(routes.ChoiceController.displayChoiceForm().url)
+        theCacheUpserted mustBe Cache(pid, DissociateUcrAnswers())
+      }
+
+      "shut MUCR" in {
+        val post = FakeRequest("POST", "/").withFormUrlEncodedBody("choice" -> Choice.ShutMUCR.value).withCSRFToken
+        val result = controller(SuccessfulAuth()).submitChoice(post)
+
+        status(result) mustBe Status.SEE_OTHER
+        redirectLocation(result) mustBe Some(routes.ChoiceController.displayChoiceForm().url)
+        theCacheUpserted mustBe Cache(pid, ShutMucrAnswers())
+      }
+    }
+
+    "return 400 when invalid" in {
```

Comment:
Intentionally *not* testing the form validation here, just testing that it gets validated